plugins/terraform/data_source_layer0_environment.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/quintilesims/layer0/cli/client"
+)
+
+func dataSourceLayer0Environment() *schema.Resource {
+	return &schema.Resource{
+		Read: datasourceLayer0EnvironmentRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"size": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"min_count": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"os": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ami": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func datasourceLayer0EnvironmentRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(client.Client)
+
+	environmentName := d.Get("name").(string)
+
+	environmentID, err := resolveTags(client, environmentName, "environment", map[string]string{})
```

Comment:
This is intentional. `resolveTags` expects a map; with `fuzz` and `type` keys being added. Are you saying I should check for a nil map in resolveTags and create one if one is not passed in?